components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineSession.kt

```diff
@@ -125,6 +125,14 @@ class SystemEngineSession(
         webView.goForward()
     }
 
+    /**
+     * See [EngineSession.goToHistoryIndex]
+     */
+    override fun goToHistoryIndex(index: Int) {
+        val historyList = webView.copyBackForwardList()
+        webView.goBackOrForward(index - historyList.currentIndex)
```

Comment:
```suggestion
        webView.goBackOrForward(index)
```
If I read [the docs](https://developer.android.com/reference/android/webkit/WebView#goBackOrForward(int)) correctly, you shouldn't need to use the currentIndex to calculate the direction, right?

Reply:
The `index` parameter corresponds to the indices in `HistoryState` which are `0..n`. So if we want to go to say index `0` when our `currentIndex = 3` we need to pass `-3`. If we don't do the math then we would call `webView.goBackOrForward(0)` which does nothing.

Reply:
I read it again and it makes sense now. I prefer the GV API more. 😄